src/subscript-loc.c

```diff
@@ -92,23 +101,43 @@ r_obj* lgl_as_location(r_obj* subscript,
   if (subscript_n == 1) {
     int elt = r_lgl_get(subscript, 0);
 
-    r_obj* out;
+    r_ssize recycle_size = n;
```

Comment:
This is specific to handling names later on. In the two cases when we return `integer()`, we recycle names to size `0` rather than size `n`

This is one of the names handling things that feels more complicated than it is worth